netCDF4/_netCDF4.pyx

```diff
@@ -1240,8 +1241,19 @@ cdef _set_att(grp, int varid, name, value, nc_type xtype=-99):
                     ierr = nc_del_att(grp._grpid, varid, attname)
                     if ierr != NC_NOERR:
                         raise RuntimeError((<char *>nc_strerror(ierr)).decode('ascii'))
-            # a unicode string, use put_att_string (if NETCDF4 file).
-            ierr = nc_put_att_string(grp._grpid, varid, attname, 1, &datstring)
+            # try to convert to ascii string, write as NC_CHAR 
+            # else it's a unicode string, write as NC_STRING (if NETCDF4)
+            try:
+                if force_ncstring: raise UnicodeDecodeError("",b"",1,1,"")
+                if python3:
+                    str(dats,encoding='ascii')
+                else:
+                    dats = dats.encode('ascii')
```

Comment:
It's suprisingly hard to write unicode-string related code that works for python 2 and 3 that doesn't look convoluted :.(